web/lib/HTTPError.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @class
+ * @param {Number}          status  HTTP Status Code
+ * @param {Error|String}    err     Original Error or Human Readable message if no error is present
+ * @param {String}          human   Human Readable Error Message
+ *
+ * @prop {Number}           status  HTTP Status Code
+ * @prop {Error}            err     Original Error Object
+ * @prop {String}           human   Human readable string
+ */
+class HTTPError {
+    constructor(status, err, human) {
+        console.error(err);
+
+        this.status = status;
+
+        if (err instanceof Error) {
+            this.err = err;
+        }
+
+        if (typeof err === 'string') {
```

Comment:
I don't follow the logic order here - I would expect to be able to call ``new HTTPError(418, err, 'I didn't ask for coffee')``, and if ``err`` is a string this will discard ``'I didn't ask for coffee'``